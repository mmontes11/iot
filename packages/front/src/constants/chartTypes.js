export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
