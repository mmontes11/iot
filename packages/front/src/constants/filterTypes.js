export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
