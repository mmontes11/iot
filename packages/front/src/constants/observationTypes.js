export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
