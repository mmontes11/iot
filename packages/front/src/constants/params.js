export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
