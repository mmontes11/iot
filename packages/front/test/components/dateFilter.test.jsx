import React from "react";
import { Provider } from "react-redux";
import DateFilter from "components/dateFilter";
import IntlProvider from "containers/intlProvider";
import { shallow, mount } from "enzyme";
import { defaultStore } from "../constants";

describe("components/dateFilter", () => {
  it("renders DateFilter for selecting a time period", () => {
    const dateFilter = {
      selector: {
        isCustomSelected: false,
        onChange: () => undefined,
      },
      timePeriod: {
        label: "Select time period:",
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete: () => undefined,
    };
    const wrapper = shallow(
      <Provider store={defaultStore}>
        <IntlProvider store={defaultStore}>
          <DateFilter dateFilter={dateFilter} />
        </IntlProvider>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders DateFilter for selecting a custom time period", () => {
    const dateFilter = {
      selector: {
        isCustomSelected: true,
        onChange: () => undefined,
      },
      timePeriod: {
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete: () => undefined,
    };
    const wrapper = shallow(
      <Provider store={defaultStore}>
        <IntlProvider store={defaultStore}>
          <DateFilter dateFilter={dateFilter} />
        </IntlProvider>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates clicks on buttons", () => {
    const onSelectorChange = jest.fn();
    const onDelete = jest.fn();
    const dateFilter = {
      selector: {
        isCustomSelected: true,
        onChange: onSelectorChange,
      },
      timePeriod: {
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete,
    };
    const wrapper = mount(
      <Provider store={defaultStore}>
        <IntlProvider store={defaultStore}>
          <DateFilter dateFilter={dateFilter} />
        </IntlProvider>
      </Provider>,
    );
    wrapper.find("#time-period-button").simulate("click");
    expect(onSelectorChange).toHaveBeenCalled();
    wrapper.find("#delete-button").simulate("click");
    expect(onDelete).toHaveBeenCalled();
  });
});
