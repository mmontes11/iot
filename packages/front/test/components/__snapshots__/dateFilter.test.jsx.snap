// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dateFilter renders DateFilter for selecting a custom time period 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <InjectIntl(DateFilter)
    dateFilter={
      Object {
        "custom": Object {
          "endDate": Object {
            "onChange": [Function],
            "selected": null,
          },
          "startDate": Object {
            "onChange": [Function],
            "selected": null,
          },
        },
        "onDelete": [Function],
        "selector": Object {
          "isCustomSelected": true,
          "onChange": [Function],
        },
        "timePeriod": Object {
          "isActive": true,
          "isDisabled": false,
          "isLoading": false,
          "items": Array [
            "foo",
            "bar",
          ],
          "onButtonClick": [Function],
          "onItemClick": [Function],
          "selectedItem": null,
        },
      }
    }
  />
</Connect(IntlProvider)>
`;

exports[`components/dateFilter renders DateFilter for selecting a time period 1`] = `
<Connect(IntlProvider)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <InjectIntl(DateFilter)
    dateFilter={
      Object {
        "custom": Object {
          "endDate": Object {
            "onChange": [Function],
            "selected": null,
          },
          "startDate": Object {
            "onChange": [Function],
            "selected": null,
          },
        },
        "onDelete": [Function],
        "selector": Object {
          "isCustomSelected": false,
          "onChange": [Function],
        },
        "timePeriod": Object {
          "isActive": true,
          "isDisabled": false,
          "isLoading": false,
          "items": Array [
            "foo",
            "bar",
          ],
          "label": "Select time period:",
          "onButtonClick": [Function],
          "onItemClick": [Function],
          "selectedItem": null,
        },
      }
    }
  />
</Connect(IntlProvider)>
`;
